### PR TITLE
Fix CDN container population when iterating

### PR DESCRIPTION
**Pull request by jamiehannaford**

### CODE REVIEW

File: lib/OpenCloud/ObjectStore/Service.php [hunk 1]
```diff
@@ -67,6 +67,18 @@ public function getCdnService()
     }
 
     /**
+     * List all available containers.
+     *
+     * @param array $filter
+     * @return \OpenCloud\Common\Collection\PaginatedIterator
+     */
+    public function listContainers(array $filter = array())
```
Comment (ycombinator) [inline lib/OpenCloud/ObjectStore/Service.php]: Thank you for refactoring this down. It was silly to have a base class with conditional logic depending on the type of the instance.

File: lib/OpenCloud/ObjectStore/CDNService.php [hunk 2]
```diff
@@ -24,4 +27,36 @@ class CDNService extends AbstractService
...
+     * @param array $filter
+     * @return \OpenCloud\Common\Collection\PaginatedIterator
+     */
+    public function listContainers(array $filter = array())
+    {
+        $filter['format'] = 'json';
+        return $this->resourceList('CDNContainer', $this->getUrl(null, $filter), $this);
+    }
+
+    public function cdnContainer($data)
```
Comment (ycombinator) [inline lib/OpenCloud/ObjectStore/CDNService.php]: Does this need to be public?

Comment (jamiehannaford) [inline lib/OpenCloud/ObjectStore/CDNService.php]: yes because the iterator is an external class

**Comment by ycombinator** [comment]
I don't see any mention of `CDNService` in the [ObjectStore documentation](https://github.com/rackspace/php-opencloud/tree/working/docs/userguide/ObjectStore). How and when is a user supposed to instantiate this service?

File: lib/OpenCloud/ObjectStore/CDNService.php (hunk 2 above)
Comment (ycombinator) [inline lib/OpenCloud/ObjectStore/CDNService.php]: Ah okay. I remember now. Thanks.

**Comment by jamiehannaford** [comment]
@ycombinator I don't think there is any :disappointed: I've just added a quick reference

**Comment by ycombinator** [comment]
Thanks @jamiehannaford. I've also created https://github.com/rackspace/php-opencloud/issues/528 to port your quick reference into the user guide + create code sample(s).

**Pull request merged**